Mathlib/Combinatorics/SimpleGraph/Copy.lean

```diff
@@ -356,6 +360,205 @@ alias ⟨IsIndContained.exists_iso_subgraph, IsIndContained.of_exists_iso_subgra
+    { verts := .univ
+      Adj := ⊥
+      adj_sub := False.elim
+      edge_vert := False.elim }
+  simp only [eq_singleton_iff_unique_mem, mem_filter, mem_univ, Subgraph.coe_bot, true_and,
+    Nonempty.forall]
+  refine ⟨⟨⟨(Equiv.Set.univ _).symm, by simp⟩⟩, fun H' e ↦
+    Subgraph.ext ((set_fintype_card_eq_univ_iff _).1 <| Fintype.card_congr e.toEquiv.symm) ?_⟩
+  ext a b
+  simp only [Prop.bot_eq_false, Pi.bot_apply, iff_false]
+  exact fun hab ↦ e.symm.map_rel_iff.2 hab.coe
+
+@[simp] lemma copyCount_of_isEmpty [IsEmpty W] (G : SimpleGraph V) (H : SimpleGraph W) :
+    G.copyCount H = 1 := by
+  rw [copyCount]
+  convert card_singleton (⊥ : G.Subgraph)
+  simp only [eq_singleton_iff_unique_mem, mem_filter, mem_univ, Subgraph.coe_bot, true_and,
+    Nonempty.forall, Subsingleton.elim H ⊥]
+  haveI : IsEmpty (⊥ : G.Subgraph).verts := by simp
+  refine ⟨⟨⟨⟨isEmptyElim, isEmptyElim, isEmptyElim, isEmptyElim⟩, fun {a} ↦ isEmptyElim a⟩⟩,
+    fun H' e ↦ Subgraph.ext ?_ <| funext₂ fun a b ↦ ?_⟩
+  · simpa [Set.eq_empty_iff_forall_not_mem, filter_eq_empty_iff, ‹IsEmpty W›] using
+      e.toEquiv.symm.isEmpty_congr
+  · simp only [Subgraph.not_bot_adj, eq_iff_iff, iff_false]
+    exact fun hab ↦ e.symm.map_rel_iff.2 hab.coe
+
+@[simp] lemma copyCount_eq_zero : G.copyCount H = 0 ↔ ¬ H ⊑ G := by
+  simp [copyCount, -nonempty_subtype, isContained_iff_exists_iso_subgraph, card_pos,
+    filter_eq_empty_iff]
+
+@[simp] lemma copyCount_pos : 0 < G.copyCount H ↔ H ⊑ G := by
+  simp [copyCount, -nonempty_subtype, isContained_iff_exists_iso_subgraph, card_pos,
+    filter_nonempty_iff]
+
+end CopyCount
+
+section LabelledCopyCount
+variable [Fintype V] [Fintype W]
+
+/-- `G.labelledCopyCount H` is the number of labelled copies of `H` in `G`. See
+`SimpleGraph.copyCount` for the number of unlabelled copies. -/
+noncomputable def labelledCopyCount (G : SimpleGraph V) (H : SimpleGraph W) : ℕ := by
+  classical exact Fintype.card (Copy H G)
+
+@[simp] lemma labelledCopyCount_of_isEmpty [IsEmpty W] (G : SimpleGraph V) (H : SimpleGraph W) :
+    G.labelledCopyCount H = 1 := by
+  convert Fintype.card_unique
+  exact { default := ⟨default, isEmptyElim⟩, uniq := fun _ ↦ Subsingleton.elim _ _ }
+
+@[simp] lemma labelledCopyCount_eq_zero : G.labelledCopyCount H = 0 ↔ ¬ H ⊑ G := by
+  simp [labelledCopyCount, IsContained, Fintype.card_eq_zero_iff, isEmpty_subtype]
+
+@[simp] lemma labelledCopyCount_pos : 0 < G.labelledCopyCount H ↔ H ⊑ G := by
+  simp [labelledCopyCount, IsContained, Fintype.card_pos_iff]
+
+/-- There's more labelled copies of `H` in `G` than unlabelled ones. -/
+lemma copyCount_le_labelledCopyCount : G.copyCount H ≤ G.labelledCopyCount H := by
+  classical
+  rw [copyCount, ← Fintype.card_coe]
+  refine Fintype.card_le_of_surjective (fun f ↦ ⟨f.toSubgraph, ?_⟩) ?_
+  · simpa using ⟨f.isoToSubgraph⟩
+  · simpa [Surjective, -Copy.range_toSubgraph] using fun H' e ↦ Copy.toSubgraph_surjOn ⟨e⟩
+
+end LabelledCopyCount
+
+/-!
+#### Induced copies
+
+TODO
```

Comment:
Consider making this a module-level TODO instead

Reply:
I've made it a TODO in the module doc, but kept it there too